src/xds/client.rs

```diff
@@ -628,9 +630,17 @@ impl AdsClient {
                 .await?,
         )?;
 
+        let mut req = tonic::Request::new(outbound);
+        self.config.xds_headers.iter().for_each(|(k, v)| {
+            let key: tonic::metadata::MetadataKey<_> = k.as_str().parse().unwrap();
```

Comment:
It would probably be good to not panic on error. Either return an error here, or maybe we can make the `xds_headers` object a `Vec<(MetadataKey, MetadataValue)>` or similar upfront so we fail at config construction time instead of down the line

Reply:
Thank you for the comment!

I tried to use `Vec<(AsciiMetadataKey, AsciiMetadataValue)>` but both `AsciiMetadataKey` and `AsciiMetadataValue` do not implement the `serde::Serialize` trait.

So at the end I created a wrapper struct called `MetadataVector`, which implements serde::Serialize` trait.